starknet_py/net/models/abi/definition.py

```diff
@@ -0,0 +1,180 @@
+        structs: Dict[str, StructType] = {}
+
+        # Example problem (with a simplified json structure):
+        # [{name: User, fields: {id: Uint256}}, {name: "Uint256", ...}]
+        # User refers to Uint256 even though it is not known yet (will be parsed next).
+        # This is why it is important to create the structure types first. This way other types can already refer to
+        # them when parsing types, even thought their fields are not filled yet.
+        # At the end we will mutate those structures to contain the right fields. An alternative would be to use
+        # topological sorting with an additional "unresolved type", so this flow is much easier.
+        for name, struct in structs_dict.items():
+            structs[name] = StructType(name, OrderedDict())
+            struct_members[name] = sorted(struct["members"], key=lambda m: m["offset"])
+
+        # Now parse the types of members and save them.
+        type_parser = TypeParser(structs)
+        for name, struct in structs.items():
+            members = AbiDefinition._parse_members(
+                f"members of structure '{name}'",
+                # pyright can't handle list of StructMemberDict here, even though TypedMemberDict
+                # is parent of StructMemberDict
+                cast(List[TypedMemberDict], struct_members[name]),
+                type_parser,
+            )
+            struct.types.update(members)
+
+        # All types have their members assigned now
+        return structs
+
+    @staticmethod
+    def _parse_abi_function(
+        function: FunctionDict, type_parser: TypeParser
+    ) -> AbiDefinition.Function:
+        return AbiDefinition.Function(
+            name=function["name"],
+            inputs=AbiDefinition._parse_members(
+                function["name"], function["inputs"], type_parser
+            ),
+            outputs=AbiDefinition._parse_members(
+                function["name"], function["outputs"], type_parser
+            ),
+        )
+
+    @staticmethod
+    def _parse_event(event: EventDict, type_parser: TypeParser) -> AbiDefinition.Event:
+        return AbiDefinition.Event(
+            name=event["name"],
+            data=AbiDefinition._parse_members(
+                event["name"], event["data"], type_parser
+            ),
+        )
+
+    @staticmethod
+    def _parse_members(
+        entity_name: str, params: List[TypedMemberDict], parser: TypeParser
+    ) -> OrderedDict[str, CairoType]:
+        # Without cast it complains that
+        # 'Type "TypedMemberDict" cannot be assigned to type "T@_group_by_name"'
+        members = AbiDefinition._group_by_name(entity_name, cast(List[Dict], params))
+        return OrderedDict(
+            (name, parser.parse_inline_type(param["type"]))
+            for name, param in members.items()
+        )
+
+    @staticmethod
+    def _group_by_name(entity_name: str, dicts: List[T]) -> OrderedDict[str, T]:
```

Comment:
I'd have the parameter `entity_name` follow the `dicts` parameter. Having it as a primary one suggest that it is the name that is somehow used in the grouping process, at least that's what I though at first.

And in fact it's pretty much only used for having better error messages.

Reply:
Also may it could be renamed? `_group_by_entry_name` or something like that?